Library/Homebrew/formula.rb

```diff
@@ -133,9 +138,14 @@ def initialize(name, path, spec)
     @path = path
     @revision = self.class.revision || 0
 
-    if path.to_s =~ HOMEBREW_TAP_PATH_REGEX
-      @full_name = "#{Tap.fetch($1, $2)}/#{name}"
+    if path == Formulary.core_path(name)
+      @tap = CoreTap.instance
+      @full_name = name
+    elsif path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+      @tap = Tap.fetch($1, $2)
```

Comment:
At some point (maybe not this PR) I like the idea of abstracting the path regex and $1/$2 into a function that returns them given a path.